### PR TITLE
Fix massive memory leak in unit tests

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

ADP-758, #2472 

# Overview

- [x] Make sure to also close the DBLayers we create!
- [x] <s>Use validateGenerator to also test non-QSM Arbitrary instances</s>
- [ ] <s>TODO: Look a bit more at the generators / validation of them, could perhaps split the PR too.</s>

# Comments

Before this fix the memory usage would steadily climb upwards 5-15GB. Now it's at most ~40 MB.

### Running part of unit tests before fix:
ska__rmavbild_2021-02-25_kl _12 34 43

### Running all unit tests after fix:
Skärmavbild 2021-02-25 kl  15 54 16

### CODE REVIEW

File: lib/core/test/unit/Cardano/Wallet/DB/Arbitrary.hs [hunk 1]
```diff
@@ -442,8 +446,18 @@ instance (Ord a, Arbitrary a) => Arbitrary (Range a) where
...
 
+
+twentyNineNulls :: BS.ByteString
+twentyNineNulls = "\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL"
+
+sixteenNulls :: BS.ByteString
+sixteenNulls = "\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL"
+
+sixteenNullsHex :: Hash purpose
+sixteenNullsHex = Hash . convertToBase Base16 $ sixteenNulls
```
Comment (Anviking) [inline lib/core/test/unit/Cardano/Wallet/DB/Arbitrary.hs]: Should look at reverting this now, I think

Comment (Anviking) [reply lib/core/test/unit/Cardano/Wallet/DB/Arbitrary.hs]: Dropped.

**Review by rvl — APPROVED** [review]
Good stuff

**Pull request merged**